OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs

```diff
@@ -41,18 +41,29 @@ public class ProductionQueueInfo : ITraitInfo
 		[Desc("This percentage value is multiplied with actor cost to translate into build time (lower means faster).")]
 		public readonly int BuildDurationModifier = 100;
 
+		[Desc("Maximum number of a single actor type that can be queued (0 = infinite).")]
+		public readonly int ItemLimit = 999;
+
+		[Desc("Maximum number of items that can be queued across all actor types (0 = infinite).")]
+		public readonly int QueueLimit = 0;
+
 		[Desc("The build time is multiplied with this value on low power.")]
 		public readonly int LowPowerSlowdown = 3;
 
 		[Desc("Notification played when production is complete.",
 			"The filename of the audio is defined per faction in notifications.yaml.")]
 		public readonly string ReadyAudio = "UnitReady";
 
-		[Desc("Notification played when you can't train another unit",
+		[Desc("Notification played when you can't train another actor",
 			"when the build limit exceeded or the exit is jammed.",
 			"The filename of the audio is defined per faction in notifications.yaml.")]
 		public readonly string BlockedAudio = "NoBuild";
```

Comment:
Might this cause #15071? You removed `NoBuild` from the notifications but I don't see any overwrites for `BlockedAudio`.